### DISCUSSION TITLE
Custom ElementCondition in plugin

### DISCUSSION THREAD

**Discussion by janhenckens** [craftcms/cms#15563] [Q&A] [answered]
In my [Exporter plugin](https://github.com/studioespresso/craft-exporter), I'd like to use the concept of ElementConditions to give users more control over which elements they want to export.

In the first step, I have them choose the Element type, the section/group/etc and the site. Based on that I'm trying to create an ElementCondition, but can't seem to get the correct fields to show up for the correct query.

For example, I have 2 category groups.
Group A, with a a plain Text field called "Test"
Group B, with a text-area called "Summary".

Here's how I'l creating the condition:
````php

$condition = $this->elementType::createCondition();
$condition->queryParams = ['section', 'sectionId', 'site', 'entryType'];

$name = ElementHelper::generateSlug($this->name) . '-' . $this->id;
$condition->name = "condition-{$name}";

$condition->elementType = $this->elementType;
$condition->modifyQuery($this->getElementQuery());
````

However, conditions for both groups contain the same fields, each with the option to set a condition for a field that isn't present on the element in question.

To debug I checked which elements were returned from my `$this->getElementQuery()` function and those were from the expected category group. I tried giving them unique names but the didn't change things.

Am I missing anything as to how this works? Appreciate any feedback you might have :)
Thanks
Jan

**Comment by brandonkelly** [comment, marked as answer]
You can set the `sourceKey` property on the condition (e.g. `group:<uid>`) or if you want it to include the custom fields from multiple field layouts, you can pass the field layouts to the condition via `$condition->setFieldLayouts()`.

**Reply by janhenckens** [reply]
Thanks! Still on Craft 4 for now and it looks like conditions don't have the `setFieldLayouts()` there yet - but I got it working with the sourceKey!
